File: src/main/java/com/ptit/timtro/model/Post.java
package com.ptit.timtro.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Post {
    private Integer id;
    private String title;
    private String content;
    private Integer price;
    private Integer acreage;
    private Integer view;
    private String address;
    private Boolean status;
    private Double latitude;
    private Double longitude;
    private Date createTime;
    private String phoneNumber;
    private List<Image> images;
    private User user;
    private Ward ward;
    private District district;
    private Province province;
    private Category category;
    private List<Tag> tags;
}

File: src/app/model/post.ts
import {Image} from "./image";
import {Category} from "./category";
import {Tag} from "./tag";
import {User} from "./user";
import {Favorite} from "./favorite";
import {Ward} from "./address/Ward";
import {District} from "./address/District";
import {Provinces} from "./address/Provinces";
import {Payment} from "./payment";

export class Post {
  id: number;
  title: string;
  content: string;
  price: number;
  acreage: number;
  view: number;
  address: string;
  status: boolean;
  latitude: number;
  longitude: number;
  createTime: string;
  phoneNumber: string;
  images: Array<Image>;
  category: Category;
  tags: Array<Tag>;
  user: User;
  comments: Array<Comment>;
  favorites: Array<Favorite>;
  ward: Ward;
  district: District;
  province: Provinces;
  payments: Array<Payment>;
}
